--- cpmldr.asm  2024-11-14 19:52:08.446783519 +0100
+++ cpmtst.asm  2024-11-19 14:37:43.679556356 +0100
@@ -48,7 +48,21 @@
        lxi     d,signon
        call    bdos

-       mvi     c,open$func     ;open the CPM3.SYS file
+       lda     (fcb+1)         ;cmd line arg
+       cpi     ' '             ;empty?
+       jz      opcpm3          ;-> open CPM3.SYS
+
+       lxi     d,fcb           ;copy cmd line arg
+       lxi     h,cpmfcb        ;to cpm fcb
+       mvi     c,12
+cpfcb  ldax    d
+       inx     d
+       mov     m,a
+       inx     h
+       dcr     c
+       jnz     cpfcb
+
+opcpm3 mvi     c,open$func     ;open the CPM3.SYS file
        lxi     d,cpmfcb
        call    bdos
        cpi     0ffh
